client/trino-client/src/main/java/io/trino/client/StatementClientV1.java

```diff
@@ -400,6 +403,10 @@ public boolean advance()
                 cause = e;
```

Comment:
Is it possible to configure timeout to some low value? If so we can use MockConnector to sleep then it could trigger timeout.

Reply:
Can we use mock server to test that logic? Please see `TestProgressMonitor`

Reply:
Nice, this is exactly what we need here! Thanks, I wouldn't have found it myself.

Reply:
All done! PTAL